erpnext/stock/utils.py

```diff
@@ -188,10 +191,23 @@ def get_avg_purchase_rate(serial_nos):
 
 def get_valuation_method(item_code):
 	"""get valuation method from item or default"""
-	val_method = frappe.db.get_value('Item', item_code, 'valuation_method')
+	val_method, batch_wise_valuation, has_batch_no, has_serial_no = frappe.db.get_value('Item', item_code,
+		['valuation_method', 'batch_wise_valuation', 'has_batch_no', 'has_serial_no'])
+
 	if not val_method:
 		val_method = frappe.db.get_value("Stock Settings", None, "valuation_method") or "FIFO"
-	return val_method
+
+	if not has_batch_no or has_serial_no:
+		batch_wise_valuation = 0
```

Comment:
if it does have batch no and not serial no, batch_wise_valuation must  be 1 right?

Reply:
If it does have batch no and does not have serial number than it can either use batch wise valuation (if applied for that item or globally) or the normal valuation method (for backwards compatibility)

If there is a serial no then it shouldn't use batch wise valuation but rather serial no wise valuation

Reply:
Use batch wise valuation only when has batch no, does not have serial no and has batch wise valuation setting